### PR TITLE
Convert Guava `Sets.filter`

### PR DESCRIPTION
**Pull request by lredor**
## What's changed?
- Add conversion for `Sets.filter` (as described in #897)

## Anything in particular you'd like reviewers to focus on?
First contributions, so I’m not sure I’ve mastered all aspects of OpenRewrite yet.

## Anyone you would like to review specifically?
@timtebeek 

### Checklist
- [x] I've added unit tests to cover both positive and negative cases
- [x] I've read and applied the [recipe conventions and best practices](https://docs.openrewrite.org/authoring-recipes/recipe-conventions-and-best-practices)
- [x] I've used the IntelliJ IDEA auto-formatter on affected files

### CODE REVIEW

File: src/main/java/org/openrewrite/java/migrate/guava/NoGuavaSetsFilter.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+                new JavaIsoVisitor<ExecutionContext>() {
+                    @Override
+                    public J.MethodInvocation visitMethodInvocation(J.MethodInvocation method, ExecutionContext ctx) {
+                        if (SETS_FILTER.matches(method)) {
+                            maybeRemoveImport("com.google.common.base.Predicate");
+                            maybeRemoveImport("com.google.common.collect.Sets");
+                            maybeAddImport("java.util.function.Predicate");
+                            maybeAddImport("java.util.stream.Collectors");
+
+                            return JavaTemplate.builder("#{any(java.util.Collection)}.stream().filter(#{any(java.util.function.Predicate)}).collect(Collectors.toSet())")
```
Comment (timtebeek) [inline src/main/java/org/openrewrite/java/migrate/guava/NoGuavaSetsFilter.java]: There's a slight concern here where `Sets` has special handling for `SortedSet`, whereas the the replacement always uses `toSet`, which uses `HashSet`. Should we conditionally use `.toCollection(TreeSet::new)` ?

Comment (lredor) [reply src/main/java/org/openrewrite/java/migrate/guava/NoGuavaSetsFilter.java]: You're right. I added a case for that.

File: src/test/java/org/openrewrite/java/migrate/guava/NoGuavaSetsFilterTest.java [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+              import java.util.Set;
+              import java.util.SortedSet;
+
+              import com.google.common.base.Predicate;
+              import com.google.common.collect.Sets;
+
+              class Test {
+                  public static Set<Object> test(SortedSet<Object> set, Predicate<Object> isNotNull) {
+                      Set<Object> indirectSet = set;
+                      return Sets.filter(indirectSet, isNotNull);
```
Comment (timtebeek) [inline src/test/java/org/openrewrite/java/migrate/guava/NoGuavaSetsFilterTest.java:125]: This case is tricky; we can't really deduce from the type system whether we'd end up in this conditional of special handling:
https://github.com/google/guava/blob/21cdcf6d9a49c1a498acb422471d8d71f9219865/guava/src/com/google/common/collect/Sets.java#L1190-L1194

Comment (timtebeek) [reply src/test/java/org/openrewrite/java/migrate/guava/NoGuavaSetsFilterTest.java:125]: I'd hope that to be rare enough to perhaps still merge the current work, but there is a risk here which could only be mitigated by data flow analysis to figure out if it might be a SortedSet going into this `filter(Set, Predicate)` method. 🤔

Comment (timtebeek) [reply src/test/java/org/openrewrite/java/migrate/guava/NoGuavaSetsFilterTest.java:125]: Either way thanks for the quick rework; would love to hear your thoughts on this edge case.

**Review by timtebeek — APPROVED** [review]
Thanks again! I think the risk is low here for folks relying on that `instanceof SortedSet` check, so we'll merge this as is with the test documented to fail.
